components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -79,53 +79,79 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
         /// <param name="ct"></param>
         /// <returns></returns>
         public Task<JobProcessingInstructionModel> GetAvailableJobAsync(string workerId, JobRequestModel request, CancellationToken ct = default) {
-            if (_assignedJobs.TryGetValue(workerId, out var job)) {
+            try {
+                _lock.Wait(ct);
```

Comment:
Lock should be acquired before the try block. That way, if we fail to acquire it, we will not get to try-catch-finally block and will not release it.

Reply:
good point, updated for GetAvailableJobAsync and SendHearthbeatAsync